source/util/ilist.h

```diff
@@ -95,29 +99,42 @@ class IntrusiveList {
     }
 
    private:
+    iterator_template() = delete;
     inline iterator_template(T* node) { node_ = node; }
     T* node_;
 
     friend IntrusiveList;
   };
 
-  typedef iterator_template<NodeType> iterator;
-  typedef iterator_template<const NodeType> const_iterator;
-
-  iterator begin();
-  iterator end();
-  const_iterator begin() const;
-  const_iterator end() const;
-
-  // Appends |node| to the end of the list.
+  using iterator = iterator_template<NodeType>;
+  using const_iterator = iterator_template<const NodeType>;
+
+  // Various types of iterators for the start (begin) and one past the end (end)
+  // of the list.
+  //
+  // Decrementing |end()| iterator will give and iterator pointing to the last
+  // element in the list, if one exists.
+  //
+  // Incrementing |end()| iterator will give |begin()|.
```

Comment:
This wraparound behaviour is good to document as an internal behaviour, as it makes operation very clear.
Hopefully nobody relies on it.  It's an implementation detail.  I'm tempted to say that here.  But this feedback is below the bar.